packages/cloud_firestore/cloud_firestore/lib/src/query.dart

```diff
@@ -603,7 +607,13 @@ class _JsonQuery implements Query<Map<String, dynamic>> {
     Iterable<Object?>? whereNotIn,
     bool? isNull,
   }) {
-    _assertValidFieldType(field);
+    _assertValidFieldType(fieldOrFilter);
+
+    if (fieldOrFilter is Filter) {
```

Comment:
Shouldn't there be an assert which prevents the usage of both Filter + isEqualTo/co? It sounds redundant

Reply:
Nice catch, I've added it :)